rsrcdump/extract.py

```diff
@@ -1,16 +1,24 @@
+from typing import Dict, List, Union
+
 import os
 import base64
 import json
+
 from rsrcdump.resconverters import converters
 from rsrcdump.textio import sanitize_type_name, sanitize_resource_name
+from rsrcdump.resfork import Resource
 
-def extract_resource_map(res_map, outpath, include_types=[], exclude_types=[], quiet=False):
+def extract_resource_map(res_map: Dict[bytes, Dict[int, Resource]],
+                         outpath: str,
+                         include_types: List[bytes]=[],
+                         exclude_types: List[bytes]=[],
+                         quiet: bool=False) -> None:
     try:
         os.mkdir(outpath)
     except FileExistsError:
         pass
 
-    J = {}
+    J: Dict[str, Dict[int, Dict[str, Union[int, str]]]] = {}
```

Comment:
```suggestion
    J: dict[str, dict[int, dict[str, Any]]] = {}
```
See below for an explanation